openscm/adapters/dice.py

```diff
@@ -0,0 +1,388 @@
+"""
+Adapter for the climate component from the Dynamic Integrated Climate-Economy (DICE)
+model by William Nordhaus.
+
+Original source: https://sites.google.com/site/williamdnordhaus/dice-rice
+"""
+
+from collections import namedtuple
+from datetime import datetime
+from math import log2
+from typing import Any
+
+import numpy as np
+
+from ..adapter import Adapter
+from ..parameters import ParameterType
+from ..timeseries_converter import (
+    ExtrapolationType,
+    InterpolationType,
+    create_time_points,
+)
+
+YEAR = 365 * 24 * 60 * 60
+
+MODEL_PARAMETER_DEFAULTS = {
+    # Initial concentration in atmosphere
+    "M_atm0": (830.4, "GtC"),
+    # Equilibrium concentration atmosphere
+    "M_atm_eq": (588, "GtC"),
+    # Lower bound concentration in atmosphere
+    "M_atm_lower": (10, "GtC"),
+    # Initial concentration in lower strata
+    "M_l0": (10010, "GtC"),
+    # Equilibrium concentration in lower strata
+    "M_l_eq": (10000, "GtC"),
+    # Lower bound for concentration in lower strata
+    "M_l_lower": (1000, "GtC"),
+    # Initial concentration in upper strata
+    "M_u0": (1527, "GtC"),
+    # Equilibrium concentration in upper strata
+    "M_u_eq": (1350, "GtC"),
+    # Lower bound for concentration in upper strata
+    "M_u_lower": (100, "GtC"),
+    # Initial atmospheric temperature change (rel. to 1900)
+    "T_atm0": (0.8, "degC"),
+    # Upper bound for atmospheric temperature change (rel. to 1900)
+    "T_atm_upper": (40, "degC"),
+    # Initial lower stratum temperature change (rel. to 1900)
+    "T_ocean0": (0.0068, "degC"),
+    # Lower bound for lower stratum temperature change (rel. to 1900)
+    "T_ocean_lower": (-1, "degC"),
+    # Upper bound for lower stratum temperature change (rel. to 1900)
+    "T_ocean_upper": (20, "degC"),
+    # Carbon cycle transition matrix
+    "b12": (0.0181, ""),
+    # Carbon cycle transition matrix
+    "b23": (0.00071, ""),
+    # Climate equation coefficient for upper level
+    "c1": (0.0222, "degC*m^2/W"),
+    # Transfer coefficient upper to lower stratum
+    "c3": (0.09175, "W/m^2/degC"),
+    # Transfer coefficient for lower level
+    "c4": (0.00487, ""),
+    # Forcings of equilibrium CO2 doubling
+    "fco22x": (3.8, "W/m^2"),
+    # 2010 forcings of non-CO2 GHG
+    "fex0": (0.25, "W/m^2"),
+    # 2100 forcings of non-CO2 GHG
+    "fex1": (0.7, "W/m^2"),
+    # Equilibrium climate sensitivity
+    "t2xco2": (2.9, "degC"),
+}
+
+
+class DICE(Adapter):
+    """
+    Adapter for the climate component from the Dynamic Integrated Climate-Economy (DICE)
+    model.
+
+    TODO recalculate increase to absolute values
+    TODO still calibrated to initial year 2010
+    TODO What about timeseries parameters when run times change?
+    """
+
+    _period_length: int
+    """Period length in seconds (currently 1yr)"""
```

Comment:
```suggestion
    """Period length in seconds (default 1yr)"""
```

?

Reply:
As in the period length could be set by the user so 'currently' doesn't really make any sense. It is by default 1yr, but if the user really forced things, they could change it. Maybe deleting 'currently' would also be fine, but the word 'currently' here implies that it might have been something else in the past/future so is just confusing.

Reply:
yes, it’s meant to be changable in the future. will add a TODO instead

Reply:
ok an in future 1yr would be the default value?

Reply:
yes